src/main/java/edu/kit/ipd/crowdcontrol/objectservice/crowdworking/PlatformManager.java

```diff
@@ -0,0 +1,240 @@
+                        throw new IllegalStateException("Updating record for published task failed");
+                    }
+                    return true;
+                });
+    }
+
+    /**
+     * Unpublish a given experiment from the given platform
+     *
+     * @param name The name of the platform
+     * @param experiment The experiment to unpublish
+     * @return None if the platform was not found, false if the unpublish failed and true if everything went fine
+     */
+    public CompletableFuture<Boolean> unpublishTask(String name, Experiment experiment) throws TaskOperationException {
+        TaskRecord record;
+
+        record = tasksOps.searchTask(name, experiment.getId()).orElse(null);
+
+        if (record == null)
+            return CompletableFuture.completedFuture(true);
+
+        return getPlatform(name).map(platform -> platform.unpublishTask(record.getPlatformData()))
+                .orElseThrow(() -> new IllegalArgumentException("Experiment not found!"))
+                .thenApply(aBoolean -> {
+                    record.setStatus(TaskStatus.finished);
+                    return tasksOps.updateTask(record);
+                });
+    }
+
+    /**
+     * update the given experiment on the given platform
+     * @param name The name of the platform
+     * @param experiment The experiment to update
+     * @return None if the platform was not found, false if the update failed and true if everything went fine.
+     */
+    public CompletableFuture<Boolean> updateTask(String name, Experiment experiment) throws TaskOperationException {
+        TaskRecord record;
+
+        record = tasksOps.searchTask(name, experiment.getId()).
+                orElseThrow(() -> new TaskOperationException("Experiment is not published"));
+
+        return getPlatform(name)
+                .map(platform -> platform.updateTask(record.getPlatformData(), experiment))
+                .orElseThrow(() -> new IllegalArgumentException("Platform not found"))
+                .thenApply(s -> {
+                    record.setPlatformData(s);
+                    return tasksOps.updateTask(record);
+                });
+    }
+
+    /**
+     * Parse a worker id out of the params which got passed by a platform
+     * @param name The name of the platform
+     * @param params Params passed by the platform
+     * @return A String if the platform exists
+     */
+    public String identifyWorker(String name, Map<String, String[]> params) throws UnknownWorkerException {
+        return getWorker(name).identifyWorker(params);
+    }
+
+    /**
+     * Get a worker if he exists
+     * @param name Name of the platform
+     * @param params Params passed by the platform
+     * @return A worker if one is found
+     * @throws UnknownWorkerException if the platform does not identify a worker
+     */
+    public Optional<WorkerRecord> getWorker(String name, Map<String ,String[]> params) throws UnknownWorkerException {
```

Comment:
no, the name of the execution must describe its purpose. often these exceptions get propagated through the system so you have to rely on good names to understand where the exception is coming from and what it is doing.